test/pool.test.ts

```diff
@@ -744,107 +770,154 @@ describe("ReceiverWeights", function () {
+    assert(weights[2].weightReceiver == 1);
+    assert(weights[2].weightProxy == 0);
   });
 
   it("Allows updating the last item", async function () {
     const weights_test = await deployReceiverWeightsTest();
     const [addr1, addr2, addr3] = randomAddresses();
 
     await weights_test.setWeights([
-      {receiver: addr1, weight: 1},
-      {receiver: addr2, weight: 2},
-      {receiver: addr3, weight: 4},
-      {receiver: addr1, weight: 8},
+      {receiver: addr1, weightReceiver: 1, weightProxy: 0},
+      {receiver: addr2, weightReceiver: 2, weightProxy: 0},
+      {receiver: addr3, weightReceiver: 4, weightProxy: 0},
+      {receiver: addr1, weightReceiver: 8, weightProxy: 0},
     ]);
 
-    assert(
-      (await weights_test.receiverWeightsSumDelta()).eq(1 + 2 + 4 - 1 + 8)
-    );
+    assert((await weights_test.weightReceiverSumDelta()).eq(1 + 2 + 4 - 1 + 8));
+    assert((await weights_test.weightProxySumDelta()).eq(0));
     const weights = await weights_test.getReceiverWeightsIterated();
     assert(weights.length == 3);
     assert(weights[0].receiver == addr3);
-    assert(weights[0].weight == 4);
+    assert(weights[0].weightReceiver == 4);
+    assert(weights[0].weightProxy == 0);
     assert(weights[1].receiver == addr2);
-    assert(weights[1].weight == 2);
+    assert(weights[1].weightReceiver == 2);
+    assert(weights[1].weightProxy == 0);
     assert(weights[2].receiver == addr1);
-    assert(weights[2].weight == 8);
+    assert(weights[2].weightReceiver == 8);
+    assert(weights[2].weightProxy == 0);
   });
 
   it("Rejects setting weight for address 0", async function () {
     const weights_test = await deployReceiverWeightsTest();
     await expectSetWeightsWithInvalidAddressReverts(weights_test, [
-      {receiver: numberToAddress(0), weight: 1},
+      {receiver: numberToAddress(0), weightReceiver: 1, weightProxy: 0},
     ]);
   });
 
   it("Rejects setting weight for address 1", async function () {
     const weights_test = await deployReceiverWeightsTest();
     await expectSetWeightsWithInvalidAddressReverts(weights_test, [
-      {receiver: numberToAddress(1), weight: 1},
+      {receiver: numberToAddress(1), weightReceiver: 1, weightProxy: 0},
     ]);
   });
+
+  it("Keeps items with only proxy weights set", async function () {
+    const weights_test = await deployReceiverWeightsTest();
+    const [addr1] = randomAddresses();
+
+    await weights_test.setWeights([
+      {receiver: addr1, weightReceiver: 0, weightProxy: 1},
+    ]);
+
+    assert((await weights_test.weightReceiverSumDelta()).eq(0));
+    assert((await weights_test.weightProxySumDelta()).eq(1));
+    const weights = await weights_test.getReceiverWeightsIterated();
+    assert(weights.length == 1);
+    assert(weights[0].receiver == addr1);
+    assert(weights[0].weightReceiver == 0);
+    assert(weights[0].weightProxy == 1);
+  });
+
+  it("Allows removing items with only proxy weights set", async function () {
```

Comment:
Given the increased size of the ReceiverWeights tests it might make sense to move them into their own test file.

Reply:
Great idea, fixed